pkg/config/config.go

```diff
@@ -383,31 +421,37 @@ func (c *Config) ToEnvVarApplyConfiguration() []*applycorev1.EnvVarApplyConfigur
 
 	for _, k := range keys {
 		envVars = append(envVars, applycorev1.EnvVar().
-			WithName(ToEnvVarName(c.SpiceConfig.EnvPrefix, k)).WithValue(c.Passthrough[k]))
+			WithName(toEnvVarName(c.SpiceConfig.EnvPrefix, k)).WithValue(c.Passthrough[k]))
 	}
 
 	return envVars
 }
 
-func (c *Config) OwnerRef() *applymetav1.OwnerReferenceApplyConfiguration {
+func (c *Config) ownerRef() *applymetav1.OwnerReferenceApplyConfiguration {
 	return applymetav1.OwnerReference().
 		WithName(c.Name).
 		WithKind(v1alpha1.SpiceDBClusterKind).
 		WithAPIVersion(v1alpha1.SchemeGroupVersion.String()).
 		WithUID(types.UID(c.UID))
 }
 
-func (c *Config) ServiceAccount() *applycorev1.ServiceAccountApplyConfiguration {
+func (c *Config) serviceAccount() *applycorev1.ServiceAccountApplyConfiguration {
 	return applycorev1.ServiceAccount(c.ServiceAccountName, c.Namespace).
 		WithLabels(metadata.LabelsForComponent(c.Name, metadata.ComponentServiceAccountLabel)).
 		WithAnnotations(c.ExtraServiceAccountAnnotations).
-		WithOwnerReferences(c.OwnerRef())
+		WithOwnerReferences(c.ownerRef())
 }
 
-func (c *Config) Role() *applyrbacv1.RoleApplyConfiguration {
+func (c *Config) ServiceAccount() *applycorev1.ServiceAccountApplyConfiguration {
+	sa := c.serviceAccount()
+	_ = ApplyPatches("ServiceAccount", sa, c.Patches)
```

Comment:
it would seem like we are applying the patch 2 times? one in `NewConfig` and one in the individual resource methods?

Reply:
in `NewConfig` it's just for validation so that we can show good errors, the actual application happens later. if we decide to go this route, I will clean it up so that it only processes once.